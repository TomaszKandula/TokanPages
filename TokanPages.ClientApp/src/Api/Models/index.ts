export type { AddArticleDto } from "./Articles/addArticleDto";
export type { UpdateArticleContentDto } from "./Articles/updateArticleContentDto";
export type { UpdateArticleCountDto } from "./Articles/updateArticleCountDto";
export type { UpdateArticleLikesDto } from "./Articles/updateArticleLikesDto";
export type { UpdateArticleVisibilityDto } from "./Articles/updateArticleVisibilityDto";
export type { RemoveArticleDto } from "./Articles/removeArticleDto";

export type { SendMessageDto } from "./Mailer/sendMessageDto";
export type { SendNewsletterDto } from "./Mailer/sendNewsletterDto";
export type { SubscriberInfoDto } from "./Mailer/subscriberInfoDto";
export type { VerifyEmailAddressDto } from "./Mailer/verifyEmailAddressDto";

export type { AddSubscriberDto } from "./Subscribers/addSubscriberDto";
export type { UpdateSubscriberDto } from "./Subscribers/updateSubscriberDto";
export type { RemoveSubscriberDto } from "./Subscribers/removeSubscriberDto";

export type { UserDataDto } from "./Users/userDataDto";
export type { ActivateUserDto } from "./Users/activateUserDto";
export type { AddUserDto } from "./Users/addUserDto";
export type { UpdateUserDto } from "./Users/updateUserDto";
export type { RemoveUserDto } from "./Users/removeUserDto";
export type { AuthenticateUserDto } from "./Users/authenticateUserDto";
export type { AuthenticateUserResultDto } from "./Users/authenticateUserResultDto";
export type { ReAuthenticateUserDto } from "./Users/reAuthenticateUserDto";
export type { ReAuthenticateUserResultDto } from "./Users/reAuthenticateUserResultDto";
export type { RevokeUserRefreshTokenDto } from "./Users/revokeUserRefreshTokenDto";
export type { ResetUserPasswordDto } from "./Users/resetUserPasswordDto";
export type { UpdateUserPasswordDto } from "./Users/updateUserPasswordDto";
export type { UserPermissionDto } from "./Users/userPermissionDto";
export type { UserRoleDto } from "./Users/userRoleDto";
export type { UploadUserMediaDto } from "./Users/uploadUserMediaDto";
export type { UploadUserMediaResultDto } from "./Users/uploadUserMediaResultDto";

export type { ErrorDto } from "./Error/errorDto";
export type { ValidationErrorsDto } from "./Error/validationErrorsDto";

export type { DocumentContentDto } from "./Components/documentContentDto";

export type { NavigationContentDto } from "./Components/navigationContentDto";
export type { HeaderContentDto } from "./Components/headerContentDto";
export type { FooterContentDto } from "./Components/footerContentDto";
export type { ClientsContentDto } from "./Components/clientsContentDto";
export type { ArticleFeaturesContentDto } from "./Components/articleFeaturesContentDto";
export type { ContactFormContentDto } from "./Components/contactFormContentDto";
export type { CookiesPromptContentDto } from "./Components/cookiesPromptContentDto";
export type { FeaturedContentDto } from "./Components/featuredContentDto";
export type { FeaturesContentDto } from "./Components/featuresContentDto";
export type { NewsletterContentDto } from "./Components/newsletterContentDto";
export type { ResetPasswordContentDto } from "./Components/resetPasswordContentDto";
export type { UpdatePasswordContentDto } from "./Components/updatePasswordContentDto";
export type { UserSigninContentDto } from "./Components/userSigninContentDto";
export type { UserSignoutContentDto } from "./Components/userSignoutContentDto";
export type { UserSignupContentDto } from "./Components/userSignupContentDto";
export type { TestimonialsContentDto } from "./Components/testimonialsContentDto";
export type { UnsubscribeContentDto } from "./Components/unsubscribeContentDto";
export type { ActivateAccountContentDto } from "./Components/activateAccountContentDto";
export type { UpdateSubscriberContentDto } from "./Components/updateSubscriberContentDto";
export type { WrongPagePromptContentDto } from "./Components/wrongPagePromptContentDto";
export type { 
    AccountContentDto, 
    SectionAccessDenied,
    SectionAccountInformation,
    SectionAccountPassword,
    SectionAccountRemoval
} from "./Components/accountContentDto";

export type { IconDto } from "./Components/Common/iconDto";
export type { LinkDto } from "./Components/Common/linkDto";
export type { RowItemDto } from "./Components/Common/rowItemDto";
export type { TextItemDto } from "./Components/Common/textItemDto";
export type { ContentDto } from "./Components/Common/contentDto";

export type { GetContentManifestDto } from "./Content/getContentManifestDto";
export type { LanguageItemDto } from "./Content/Items/languageItemDto";
export type { JWT } from "./Jwt";
