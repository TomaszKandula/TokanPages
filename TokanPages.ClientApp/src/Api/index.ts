export {
    IsSuccessStatusCode,
    GetBaseHeaders,
    GetProcessedBody,
    GetProcessedHeaders,
    GetProcessedResponse,
} from "./Utilities";

export type { ExecuteApiActionResultProps, ExecuteStoreActionProps, ExecuteApiActionProps } from "./Utilities/Types";

export {
    API_BASE_URI,
    GET_ARTICLES,
    GET_ARTICLE,
    GET_ARTICLE_BY_TITLE,
    ADD_ARTICLE,
    UPDATE_ARTICLE_CONTENT,
    UPDATE_ARTICLE_COUNT,
    UPDATE_ARTICLE_LIKES,
    UPDATE_ARTICLE_VISIBILITY,
    RETRIEVE_ARTICLE_INFO,
    REMOVE_ARTICLE,
    LOG_MESSAGE,
    GET_USERS,
    GET_USER,
    GET_USER_IMAGE,
    ADD_USER,
    ACTIVATE_USER,
    UPDATE_USER,
    REMOVE_USER,
    AUTHENTICATE,
    REAUTHENTICATE,
    VERIFY_USER_EMAIL,
    REVOKE_REFRESH_TOKEN,
    REVOKE_USER_TOKEN,
    RESET_USER_PASSWORD,
    UPDATE_USER_PASSWORD,
    UPLOAD_USER_IMAGE,
    GET_NEWSLETTERS,
    GET_NEWSLETTER,
    ADD_NEWSLETTER,
    UPDATE_NEWSLETTER,
    REMOVE_NEWSLETTER,
    SEND_MESSAGE,
    SEND_NEWSLETTER,
    NOTIFY_WEB_URL,
    NOTIFICATION_STATUS,
    GET_CONTENT_MANIFEST,
    REQUEST_PAGE_DATA,
    GET_ARTICLE_MAIN_IMAGE_URL,
    GET_IMAGES_URL,
    GET_DOCUMENTS_URL,
    ARTICLE_PATH,
    GET_USER_VIDEO,
    GET_VIDEO_ASSET,
    GET_USER_NOTES,
    GET_USER_NOTE,
    ADD_USER_NOTE,
    UPDATE_USER_NOTE,
    REMOVE_USER_NOTE,
} from "./Paths";
