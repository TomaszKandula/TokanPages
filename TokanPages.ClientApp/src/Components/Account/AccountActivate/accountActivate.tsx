import * as React from "react";
import { useDispatch, useSelector } from "react-redux";
import { ApplicationState } from "../../../Store/Configuration";
import { UserActivateAction, UserNotificationAction } from "../../../Store/Actions";
import { RECEIVED_ERROR_MESSAGE } from "../../../Shared/constants";
import { OperationStatus } from "../../../Shared/enums";
import { AccountActivateView } from "./View/accountActivateView";
import { AccountActivateProps } from "./Types";
import Validate from "validate.js";

const DefaultValues = {
    type: "Unset",
    caption: "",
    text1: "",
    text2: "",
    button: "",
};

export const AccountActivate = (props: AccountActivateProps): React.ReactElement => {
    const dispatch = useDispatch();
    const hasEmptyId = Validate.isEmpty(props.id);
    const data = useSelector((state: ApplicationState) => state.contentPageData);
    const contentData = data?.components?.accountActivate;

    const onSuccess = contentData?.onSuccess;
    const onVerifying = contentData?.onVerifying;
    const onProcessing = contentData?.onProcessing;
    const onError = contentData?.onError;

    const onSuccessWithoutLock = {
        ...onSuccess,
        text1: onSuccess?.noBusinessLock.text1,
        text2: onSuccess?.noBusinessLock.text2,
    };
    const onSuccessWithLock = {
        ...onSuccess,
        text1: onSuccess?.businessLock.text1,
        text2: onSuccess?.businessLock.text2,
    };

    const userActivation = useSelector((state: ApplicationState) => state.userActivate);
    const error = useSelector((state: ApplicationState) => state.applicationError);

    const hasContentReady = !data.isLoading && data.status === OperationStatus.hasFinished;
    const hasOperationNotStarted = userActivation?.status === OperationStatus.notStarted;
    const hasOperationFinished = userActivation?.status === OperationStatus.hasFinished;
    const hasOperationError = error?.errorMessage === RECEIVED_ERROR_MESSAGE;

    const [content, setContent] = React.useState(DefaultValues);
    const [hasProgress, setHasProgress] = React.useState(true);
    const [hasRequested, setHasRequested] = React.useState(false);
    const [hasSuccess, setHasSuccess] = React.useState(false);
    const [hasError, setHasError] = React.useState(false);

    const canProceed = hasContentReady && !hasEmptyId && hasProgress;

    React.useEffect(() => {
        if (!canProceed) return;

        if (content?.type === "Unset") {
            if (props.type === "verification") {
                setContent(onVerifying);
                return;
            }

            setContent(onProcessing);
            return;
        }

        if (hasOperationError) {
            setContent(onError);
            setHasError(true);
            setHasProgress(false);
            return;
        }

        if (hasOperationNotStarted && hasProgress && !hasRequested) {
            setHasRequested(true);
            setTimeout(
                () =>
                    dispatch(
                        UserActivateAction.activate({
                            activationId: props.id,
                        })
                    ),
                1500
            );

            return;
        }

        if (hasOperationFinished) {
            dispatch(
                UserNotificationAction.notify({
                    userId: userActivation.response.userId,
                    handler: "user_activated",
                })
            );

            setHasSuccess(true);
            setHasProgress(false);

            return userActivation.response.hasBusinessLock
                ? setContent(onSuccessWithLock)
                : setContent(onSuccessWithoutLock);
        }
    }, [
        canProceed,
        props.id,
        props.type,
        content?.type,
        hasOperationError,
        hasSuccess,
        hasProgress,
        hasRequested,
        hasOperationNotStarted,
        hasOperationFinished,
        userActivation?.response.hasBusinessLock,
        userActivation?.response.userId,
        onProcessing,
        onVerifying,
        onSuccessWithLock,
        onSuccessWithoutLock,
    ]);

    return (
        <AccountActivateView
            isLoading={data?.isLoading}
            shouldFallback={hasEmptyId}
            caption={content?.caption}
            text1={content?.text1}
            text2={content?.text2}
            fallback={{
                caption: contentData?.fallback.caption,
                text: contentData?.fallback.text,
            }}
            hasProgress={hasProgress}
            hasError={hasError}
            hasSuccess={hasSuccess}
            className={props.className}
        />
    );
};
