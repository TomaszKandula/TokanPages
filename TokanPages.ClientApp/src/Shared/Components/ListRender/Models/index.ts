export type { Fields } from "./Fields";
export type { Item } from "./Item";
export type { Subitem } from "./Subitem";
