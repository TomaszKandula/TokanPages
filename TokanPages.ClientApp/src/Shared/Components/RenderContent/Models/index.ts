export type { IArticleItem } from "./ArticleItemModel";
export type { IAuthor } from "./AuthorModel";
export type { ITextObject, ITextItem, IRowItem } from "./TextModel";
