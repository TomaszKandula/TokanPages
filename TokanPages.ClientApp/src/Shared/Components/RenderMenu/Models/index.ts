export type { Fields, MenuProps } from "./Fields";
export type { Item } from "./Item";
export type { Subitem } from "./Subitem";
