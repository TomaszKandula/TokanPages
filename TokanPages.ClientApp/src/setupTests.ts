// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
