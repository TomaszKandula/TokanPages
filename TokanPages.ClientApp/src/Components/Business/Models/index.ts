import { ServiceItemDto, TechItemsDto } from "../../../Api/Models";
import { ReactChangeEvent, ReactMouseEvent } from "../../../Shared/types";

export interface MessageFormProps {
    company: string;
    firstName: string;
    lastName: string;
    email: string;
    phone: string;
    description: string;
    techStack: string[];
    services: string[];
}

export interface TechStackListProps {
    isDisabled: boolean;
    list: TechItemsDto[];
    handler: (event: ReactChangeEvent, isChecked: boolean) => void;
}

export interface ServiceItemCardProps {
    isDisabled: boolean;
    key: React.Key | null | undefined;
    value: ServiceItemDto;
    handler: (event: ReactMouseEvent) => void;
    services: string[];
}

export interface BusinessFormProps {
    hasCaption?: boolean;
    hasIcon?: boolean;
    hasShadow?: boolean;
    className?: string;
    background?: string;
}
