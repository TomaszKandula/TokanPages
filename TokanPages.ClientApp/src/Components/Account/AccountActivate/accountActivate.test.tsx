import "../../../setupTests";
import React from "react";
import * as Redux from "react-redux";
import * as Dom from "react-router-dom";
import { render } from "@testing-library/react";
import { AccountActivateContentDto } from "../../../Api/Models";
import { ContentPageData } from "../../../Store/Defaults";
import { ApplicationDefault, ApplicationState } from "../../../Store/Configuration";
import { AccountActivate } from "./accountActivate";

jest.mock("react-router-dom", () => ({
    ...(jest.requireActual("react-router-dom") as typeof Dom),
    useLocation: () => ({
        search: "localhost:3000/accountactivation/?id=dba4043c-7428-4f72-ba13-fe782c7a88fa",
    }),
}));

jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn(),
}));

describe("Test account group component: accountActivate", () => {
    const testId = "dba4043c-7428-4f72-ba13-fe782c7a88fa";
    const testContent: AccountActivateContentDto = {
        language: "eng",
        onVerifying: {
            type: "Verifying",
            caption: "Email Verification",
            text1: "Processing your account..., please wait.",
            text2: "",
            button: "",
        },
        onProcessing: {
            type: "Processing",
            caption: "Account Activation",
            text1: "Processing your account..., please wait.",
            text2: "",
            button: "",
        },
        onSuccess: {
            type: "Success",
            caption: "Account Processing",
            button: "Go to main",
            noBusinessLock: {
                text1: "Your account has been successfully processed!",
                text2: "You can now use all our services.",
            },
            businessLock: {
                text1: "Your account has been <b>partially activated</b>.",
                text2: "Please wait for our decision. You will hear from us soon.",
            },
        },
        onError: {
            type: "Error",
            caption: "Processing Error",
            text1: "Could not process your account. Unexpected error occurred.",
            text2: "Please contact IT support.",
            button: "Retry",
        },
    };

    let state: ApplicationState = ApplicationDefault;
    state.contentPageData = ContentPageData;
    state.contentPageData.components.accountActivate = testContent;

    const useSelectorMock = jest.spyOn(Redux, "useSelector");
    const useDispatchMock = jest.spyOn(Redux, "useDispatch");
    beforeEach(() => {
        useSelectorMock.mockImplementation(callback => callback(state));
        useDispatchMock.mockReturnValue(jest.fn());
    });

    it("should render correctly '<AccountActivate />' when content is loaded.", () => {
        const html = render(<AccountActivate id={testId} type="" />);
        expect(useSelectorMock).toBeCalledTimes(9);
        expect(useDispatchMock).toBeCalledTimes(3);
        expect(html).toMatchSnapshot();
    });
});
