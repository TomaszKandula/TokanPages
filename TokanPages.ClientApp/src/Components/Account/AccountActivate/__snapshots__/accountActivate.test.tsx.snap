// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test account group component: accountActivate should render correctly '<AccountActivate />' when content is loaded. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div
          class="bulma-container bulma-is-max-desktop"
        >
          <div
            class="py-6"
          >
            <div
              class="bulma-card custom-card-margins"
            >
              <div
                class="bulma-card-content"
              >
                <div
                  class="custom-card-icon-background is-flex is-justify-content-center is-align-items-center my-6"
                >
                  <div
                    class="custom-card-icon-holder is-flex is-justify-content-center is-align-items-center"
                  >
                    <div>
                      <svg
                        class="is-flex is-align-self-center custom-card-icon has-text-info"
                        role="presentation"
                        style="width: 4.5rem; height: 4.5rem;"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M13,9H11V7H13M13,17H11V11H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
                          style="fill: currentColor;"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <h2
                  class="bulma-title has-text-centered"
                />
                <div
                  class="my-5"
                >
                  <p
                    class="is-size-6 has-text-centered line-height-20"
                  />
                  <p
                    class="is-size-6 has-text-centered line-height-20"
                  />
                </div>
                <div
                  class="mb-4"
                />
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div
        class="bulma-container bulma-is-max-desktop"
      >
        <div
          class="py-6"
        >
          <div
            class="bulma-card custom-card-margins"
          >
            <div
              class="bulma-card-content"
            >
              <div
                class="custom-card-icon-background is-flex is-justify-content-center is-align-items-center my-6"
              >
                <div
                  class="custom-card-icon-holder is-flex is-justify-content-center is-align-items-center"
                >
                  <div>
                    <svg
                      class="is-flex is-align-self-center custom-card-icon has-text-info"
                      role="presentation"
                      style="width: 4.5rem; height: 4.5rem;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M13,9H11V7H13M13,17H11V11H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
                        style="fill: currentColor;"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <h2
                class="bulma-title has-text-centered"
              />
              <div
                class="my-5"
              >
                <p
                  class="is-size-6 has-text-centered line-height-20"
                />
                <p
                  class="is-size-6 has-text-centered line-height-20"
                />
              </div>
              <div
                class="mb-4"
              />
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
